gateway/src/api/latest.rs

```diff
@@ -956,12 +979,14 @@ pub(crate) struct RouterState {
     pub service: Arc<GatewayService>,
     pub sender: Sender<BoxedTask>,
     pub running_builds: Arc<Mutex<TtlCache<Uuid, ()>>>,
+    pub posthog_client: async_posthog::Client,
 }
 
 pub struct ApiBuilder {
     router: Router<RouterState>,
     service: Option<Arc<GatewayService>>,
     sender: Option<Sender<BoxedTask>>,
+    posthog_client: Option<async_posthog::Client>,
```

Comment:
nit: Since the posthog client is essentially a hardcoded global value, it could be a static variable with OnceLock. That way we don't need to pass it through all of the functions and builder. Nothing wrong with having it this way tho.

Reply:
Interesting, would it be globally available?

Reply:
It is similar to a global variable, yeah. We can merge this as is tho.